homeassistant/components/rainbird/binary_sensor.py

```diff
@@ -3,65 +3,55 @@
 
 import logging
 
-from pyrainbird import RainbirdController
-
 from homeassistant.components.binary_sensor import (
     BinarySensorEntity,
     BinarySensorEntityDescription,
 )
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-
-from . import (
-    BINARY_SENSOR_TYPES,
-    DATA_RAINBIRD,
-    RAINBIRD_CONTROLLER,
-    SENSOR_TYPE_RAINDELAY,
-    SENSOR_TYPE_RAINSENSOR,
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
 )
 
+from . import BINARY_SENSOR_TYPES
+
 _LOGGER = logging.getLogger(__name__)
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up a Rain Bird sensor."""
     if discovery_info is None:
         return
 
-    controller = hass.data[DATA_RAINBIRD][discovery_info[RAINBIRD_CONTROLLER]]
-    add_entities(
+    async_add_entities(
         [
-            RainBirdSensor(controller, description)
+            RainBirdSensor(discovery_info[description.key], description)
             for description in BINARY_SENSOR_TYPES
         ],
         True,
```

Comment:
Is True still the right value here ?

Reply:
When removed, entities are unavailable in the tests. I fought with this a bit, then left it as is since `True` here seemed to do what I was expecting. Open to suggestions for how to improve this.

Reply:
I think the answer here is the same as above. I'll leave the other comment but resolving this one.